tools/scripts/updateDocumentSnippets.py

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python3
+
+from scriptCommon import catchPath
+import os
+import subprocess
+
+# ---------------------------------------------------
+# Update code examples
+# ---------------------------------------------------
+# For info on mdsnippets, see https://github.com/SimonCropp/MarkdownSnippets
+
+# install dotnet SDK from http://go.microsoft.com/fwlink/?LinkID=798306&clcid=0x409
+# Then install MarkdownSnippets.Tool with
+#   dotnet tool install -g MarkdownSnippets.Tool
+# To update:
+#   dotnet tool update  -g MarkdownSnippets.Tool
+# To uninstall (e.g. to downgrade to a lower version)
+# dotnet tool uninstall -g MarkdownSnippets.Tool
+
+os.chdir(catchPath)
+
+subprocess.run('dotnet tool update  -g MarkdownSnippets.Tool --version 21.2.0', shell=True, check=True)
```

Comment:
This will fail if the system already has a newer version of the tool.

Reply:
True. The reason for specifying a specific version is that there is a tendency for the formatting of output to change between versions... Which can lead to difficult merges, and random changes in doc files being mixed in with code changes...

Do you know of another way to install, that would solve that problem?

If not, I'll experiment with non-global installations...

Reply:
Not sure what is the best solution, but the global install makes this situation worse. If someone already has the tool for different project, then their version must be the same or it will fail for them.

Reply:
> Not sure what is the best solution, but the global install makes this situation worse. If someone already has the tool for different project, then their version must be the same or it will fail for them.

Off the top of my head, there are at least two options - both of which need experimentation and/or more discussion:

- Enable the CI to update docs, so that users do not have to do so
- Experiment with non-global install (which I have just tried, and looks promising)

Rather than rushing to change the code now, would you consider accepting the PR for now, as a step on the way... It is at least possible to now auto-generate docs, and for anyone who has a newer install of mdsnippets, they always have the option of not updating the docs for now?